### DISCUSSION TITLE
Add truecolor support for themes with custom colors

### DISCUSSION THREAD

**Discussion by roopeshor** [d99kris/nchat#261] [Ideas]
**Description**:
When using other color schemes, except `basic-color`, the terminal shows a blue background

**How to reproduce it**:
 - Build from source as in https://github.com/d99kris/nchat?tab=readme-ov-file#build-from-source
 - `nchat --setup`
 - copy dracula color scheme: (but goes same for all except basic-color')
   - ```cp $(dirname $(which nchat))/../share/nchat/themes/dracula/* ~/.nchat/```
 - run `nchat` with and without `TERM=xterm-256color` (though this seems to have no effect)

**Screenshots**:
 `basic-color` works well
![Screenshot_20240322_223803](https://github.com/d99kris/nchat/assets/62459820/4a53b6bd-43a7-4de1-8f4a-ffb2fd086ab4)
but  `dracula` doesn't work as expected
![Screenshot_20240322_223850](https://github.com/d99kris/nchat/assets/62459820/669dec96-b2c1-4c94-838e-3977a5c0df01)

**Environment**:
- Version: 4.41
- OS / distro: Debian 12
- Terminal : Konsole

**Comment by d99kris** [comment]
Hi @Roopesh2 - thanks for reporting an issue. I will try to look into this to see whether it's something nchat can easily support, or whether I'll need to leave it out-of-scope. For time being the workarounds would be to use themes that use standard palette (i.e. not RGB 0x112233 values), or use another terminal. Based on some quick googling I believe [this other issue](https://github.com/mawww/kakoune/issues/1807) describes the problem with using custom palette in ncurses apps in Konsole.

**Comment by d99kris** [comment]
I have done some research on this, and it seems it's not trivial to support this. I will leave it out of scope for now and convert this issue to a feature request. Maybe some other developer with experience in this area like to contribute.
